.github/workflows/uploadFirtoolBinaries.yml

```diff
@@ -105,12 +106,12 @@ jobs:
             -DLLVM_PARALLEL_LINK_JOBS=1 \
             -DLLVM_FORCE_ENABLE_STATS=${{ matrix.build_config.stats }} \
             -DCIRCT_RELEASE_TAG_ENABLED=ON \
-            -DCIRCT_RELEASE_TAG=${{ github.ref_name }} \
```

Comment:
What does this do, and why does it need to change?

Reply:
It doesn't _need_ to change, but this workflow can be run on non-tag commits via `workflow_dispatch`. By setting the release tag to the `github.ref_name`, it gives you pretty useless version `CIRCT unknown git version`. By setting it to `firtool`, [the version is derived from](https://github.com/llvm/circt/blob/3965f68d8c4633ee3de0db29645699277ce5352d/cmake/modules/GenVersionFile.cmake#LL16C47-L16C78) `git describe --dirty --tags --match <CIRCT_RELEASE_TAG>*` such that it works for random commits in addition to tagged releases.

you can see an example here: https://github.com/jackkoenig/circt/actions/runs/4535349993

Reply:
won't this make the version string worse for workflows run on tagged commits?

Reply:
(and  if I'm wrong about that feel free to just merge)

Reply:
I think this actually needs to be `firtool*` or `firtool-*` or something with an asterisk. But with that, I think it will do the same as before for tagged builds, and what Jack is going for with this. On a tagged commit, it should use the tagged commit for the version (eg `firtool-1.36.0`), but on an untagged commit, it should give something indicating the number of commits since the tag and the current hash (eg `firtool-1.36.0-24-g6c2479717`).

I tried it out on my local checkout:

```
~/circt$ git describe --dirty --tags --match firtool
fatal: No names found, cannot describe anything.
~/circt$ git describe --dirty --tags --match firtool*
firtool-1.36.0-24-g6c2479717
~/circt$ git describe --dirty --tags --match firtool-1.36.0
firtool-1.36.0-24-g6c2479717
```

And if I checkout the firtool-1.36.0 tag:

```
~/circt$ git describe --dirty --tags --match firtool
fatal: No names found, cannot describe anything.
~/circt$ git describe --dirty --tags --match firtool*
firtool-1.36.0
~/circt$ git describe --dirty --tags --match firtool-1.36.0
firtool-1.36.0
```

So it shouldn't change previous behavior on tagged jobs, but it should give a more meaningful version for untagged jobs.

Reply:
The `*` is added [here](https://github.com/llvm/circt/blob/42fd93de70c325087f93ddca1f92e998a077a96e/lib/Support/CMakeLists.txt#L38) for use [here](https://github.com/llvm/circt/blob/3965f68d8c4633ee3de0db29645699277ce5352d/cmake/modules/GenVersionFile.cmake#LL16C47-L16C78). I have tested this locally and it works correct as is for both tagged and untagged commits!

Reply:
Ah great, thanks for checking.